pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -149,6 +155,9 @@ private SegmentNameGenerator getSegmentNameGenerator() {
             Boolean.parseBoolean(segmentNameGeneratorConfigs.get(EXCLUDE_SEQUENCE_ID)),
             IngestionConfigUtils.getBatchSegmentIngestionType(tableConfig),
             IngestionConfigUtils.getBatchSegmentIngestionFrequency(tableConfig), dateTimeFormatSpec);
+      case INPUT_FILE_SEGMENT_NAME_GENERATOR:
+        return new InputFileSegmentNameGenerator(segmentNameGeneratorConfigs.get(FILE_PATH_PATTERN),
```

Comment:
I feel a much simpler solution would be resolve the segment name here and use `FixedSegmentNameGenerator` instead. There is no guarantee that the input file used to generate the segment is never moved.

Reply:
1. You are correct, we actually could set up the segment name here, because there's a new SegmentNameGenerator created for each segment. But I was following the current API convention, where the `SegmentNameGenerator.generateSegmentName()` method is called with per-segment parameters. Otherwise there's no reason for it to be an interface. If you want to change how segment naming is handled as part of this PR, I could get rid of the interface, and have a single SegmentNameGenerator class with a static method that takes the naming type, the config parameter map, and returns a segment name.
2. Though I don't follow why this change would better handle movement of input files. The segment is generated once, not dynamically, and during the actual job generating segments the input files better not be moving around, as their paths are assumed to be stable for the duration of the job (e.g. with Hadoop, these paths are stored as the input data for the segment generation job).
3. Regardless, there's still the issue of how to handle segment naming based on the original input file path, which was the use case behind issue #7090. I could make it work, but it's a bit tricky to extract paths from a URI.

Reply:
My concern here is that in several use cases (realtime to offline, segment merge etc.), the segments are not generated from input files, but from other segments. Modeling input file path as a parameter in `SegmentNameGenerator.generateSegmentName()` can cause issues for these cases. Seems generating segment name from input path only apply to `SegmentGenerationTaskRunner` so I would suggest resolving the segment name there.
But this cannot solve the local copy change path problem. Another approach would be directly setting the table name within the `SegmentNameGeneratorSpec` before copying the remote file to local

Reply:
I think I addressed this w/my most recent changes.